### PR TITLE
Fixed: some touched accounts were not verified for eip161 deletion

### PR DESCRIPTION
**Pull request by zilm13**
Let's clarify
https://github.com/ethereum/EIPs/issues/161

An account changes state when:
1. it is the target or refund of a SUICIDE operation for zero or more value; 
1. it is the source or destination of a CALL operation or message-call transaction transferring zero or more value; 
1. it is the source or newly-creation of a CREATE operation or contract-creation transaction endowing zero or more value; 
1. as the block author ("miner") it is recipient of block-rewards or transaction-fees of zero or more. 

Looks like we could answer *YES* on all of this, but you'd better check too. So:
- Do we now check and merge results of all cases?
- Should we move touched accounts to ProgramResult?

### CODE REVIEW

**Comment by zilm13** [comment]
@Nashatyrev should fix #622

**Comment by Nashatyrev** [comment]
>    Do we now check and merge results of all cases?
I added merge for regular call but forgot about create. Do you mean any other cases? 

>    Should we move touched accounts to ProgramResult?
Yep probably. Also it is possible that we shouldn't add touched accounts in case of throw. But if this will work on live then it actually doesn't matter much, since all accounts are cleared now and this execution branch will not be executed anymore

**Comment by zilm13** [comment]
> I added merge for regular call but forgot about create. Do you mean any other cases?
Just asking you to check it too according to specs to clarify that we didn't forgot anything else.
> Also it is possible that we shouldn't add touched accounts in case of throw.
Yeah, that's why I've asked about ProgramResult. So I will refactor it a bit now.

**Comment by zilm13** [comment]
Please, check

File: ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 1]
```diff
@@ -465,6 +465,7 @@ public void createContract(DataWord value, DataWord memStart, DataWord memSize)
         if (!byTestingSuite())
             track.commit();
         getResult().addDeleteAccounts(result.getDeleteAccounts());
+        getResult().addTouchAccounts(result.getTouchedAccounts());
```
Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java]: M.b. make it a part of `ProgramResult.merge()` ?

**Comment by zilm13** [comment]
@Nashatyrev Updated to use merge

**Pull request merged**